src/OpenTelemetry.Instrumentation.Hangfire/Implementation/HangfireInstrumentationJobFilterAttribute.cs

```diff
@@ -0,0 +1,68 @@
+﻿// <copyright file="HangfireInstrumentationJobFilterAttribute.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Instrumentation.Hangfire.Implementation
+{
+    using System.Diagnostics;
+    using global::Hangfire.Common;
+    using global::Hangfire.Server;
+    using OpenTelemetry.Trace;
+
+    internal class HangfireInstrumentationJobFilterAttribute : JobFilterAttribute, IServerFilter
+    {
+        private const string ActivityKey = "ActivityKey";
+
+        public void OnPerforming(PerformingContext filterContext)
+        {
+            // Short-circuit if nobody is listening
+            if (!HangfireInstrumentation.ActivitySource.HasListeners())
+            {
+                return;
+            }
+
+            var activityName = $"{filterContext.BackgroundJob.Job.Type.Name}.{filterContext.BackgroundJob.Job.Method.Name}".ToLowerInvariant();
```

Comment:
It'd be good to set displayname after the activity is confirmed to be created/sampled in, to avoid this string allocation..